src/Components/QuickDateSelect.tsx

```diff
@@ -1,86 +1,77 @@
-}
-
-export const DateSelect: React.FC<Inputs> = (props) => {
-  enum menu {
-    term = 0,
-    interval = 1,
-  }
-
-  function getAnchorEl(identifier: number): Element {
-    if (identifier == menu.term) {
-      return props.termAnchorEl;
-    } else {
-      return props.intervalAnchorEl;
-    }
-  }
-
-  function setAnchorEl(identifier: number, item: EventTarget | null): void {
-    if (identifier == menu.term) {
-      props.setTermAnchorEl(item);
-    } else {
-      props.setIntervalAnchorEl(item);
-    }
-  }
-
-  function handleMenuClick(identifier: number, event: EventTarget): void {
-    setAnchorEl(identifier, event);
-  }
-
-  function handleClose(identifier: number, item: string): void {
-    setAnchorEl(identifier, null);
-    let words = props.quickSelectContent;
-    words[identifier] = item;
-    props.handleClick(words);
-  }
-
-  function getDropdownObj(index: number) {
-    return {
-      relativeTerms: props.quickSelectTerms,
-      relativeIntervals: props.quickSelectIntervals,
-      relativeSelectContent: props.quickSelectContent,
-      handleClose,
-      handleMenuClick,
-      getAnchorEl,
-      classes: props.classes,
-      identifier: index,
-    };
-  }
-
-  return (
-    <Box className={props.classes.flexRow}>
-      <Box>
-        <RelativeDateSelectDropdown {...getDropdownObj(0)} />
-      </Box>
-      <Box ml={1}>
-        <RelativeDateSelectDropdown {...getDropdownObj(1)} />
-      </Box>
-    </Box>
-  );
-};
+import React from "react";
+import { Box } from "@material-ui/core";
+import { DateRange, TermContext } from "./DateRange.tsx";
+import { RelativeDateSelectDropdown } from "./RelativeDateSelectDropdown.tsx";
+
+interface Inputs {
+  termAnchorEl: any;
+  intervalAnchorEl: any;
+  quickSelectTerms: string[];
+  quickSelectIntervals: number[];
+  quickSelectContent: number[];
+  classes: any;
+  dateRange: DateRange;
+  handleClick(text: number): void;
+  setTermAnchorEl(element: EventTarget | null): void;
+  setIntervalAnchorEl(element: EventTarget | null): void;
+}
+
+export const DateSelect: React.FC<Inputs> = (props) => {
+  enum menu {
+    term = 0,
+    interval = 1,
+  }
+
+  function getAnchorEl(identifier: number): Element {
+    if (identifier == menu.term) {
+      return props.termAnchorEl;
+    } else {
+      return props.intervalAnchorEl;
+    }
+  }
+
+  function setAnchorEl(identifier: number, item: EventTarget | null): void {
+    if (identifier == menu.term) {
+      props.setTermAnchorEl(item);
+    } else {
+      props.setIntervalAnchorEl(item);
+    }
+  }
+
+  function handleMenuClick(identifier: number, event: EventTarget): void {
+    setAnchorEl(identifier, event);
+  }
+
+  function handleClose(identifier: number, item: any): void {
+    setAnchorEl(identifier, null);
+    let words = props.quickSelectContent;
+    words[identifier] = item;
+    props.handleClick(words[0] * words[1]);
+  }
+
+  function getDropdownObj(index: number) {
```

Comment:
Instead of creating two objects like this, create one object and then in the code where you use it below do this:
`{ ...dropDownProps, identifier: 0}`  Cuts down on a few lines of code and eliminates a function call.

Reply:
I would have to pass the whole props object which the component doesnt need, and since it is used in QuickDateSelect and RelativeDateSelect, it might create continuity error regarding naming.. there are also a handful of properties othen than index that do not come from the Inputs interface (eg. HandleMenuClick, handleClose, and getAnchorEl

Reply:
Something more like this:
```js
  const dropdownProps = {
      relativeTerms: props.quickSelectTerms,
      relativeIntervals: props.quickSelectIntervals,
      firstDropdownText: props.quickSelectTerms,
      secondDropdownText: props.quickSelectIntervals,
      relativeSelectContent: props.quickSelectContent,
      dropdownType: TermContext.quickSelect,
      handleClose,
      dateRange: props.dateRange,
      handleMenuClick,
      getAnchorEl,
      classes: props.classes,
      identifier: index,
    };

    ...

    <RelativeDateSelectDropdown {... dropdownProps, identifier: 0} />
    <RelativeDateSelectDropdown {... dropdownProps, identifier: 1} />
```